app/store/actions/healthcareAuthorities/getHealthcareAuthoritiesAction.ts

```diff
@@ -0,0 +1,55 @@
+import { createAction } from '@reduxjs/toolkit';
+import Yaml from 'js-yaml';
+import { Dispatch } from 'redux';
+
+import { AUTHORITIES_LIST_URL_MVP1 } from '../../../constants/authorities';
+import { AppThunk } from '../../types';
+
+const GET_HEALTHCARE_AUTHORITIES_STARTED = 'GET_HEALTHCARE_AUTHORITIES_STARTED';
+const GET_HEALTHCARE_AUTHORITIES_SUCCESS = 'GET_HEALTHCARE_AUTHORITIES_SUCCESS';
+const GET_HEALTHCARE_AUTHORITIES_FAILURE = 'GET_HEALTHCARE_AUTHORITIES_FAILURE';
+
+const getHealthcareAuthorities_started = createAction(
+  GET_HEALTHCARE_AUTHORITIES_STARTED,
+);
+const getHealthcareAuthorities_success = createAction(
+  GET_HEALTHCARE_AUTHORITIES_SUCCESS,
+  ({ healthcareAuthorities }) => ({ payload: { healthcareAuthorities } }),
+);
+const getHealthcareAuthorities_failure = createAction(
+  GET_HEALTHCARE_AUTHORITIES_FAILURE,
+  ({ error }) => ({ payload: {}, meta: { error } }),
+);
+
+const getHealthcareAuthoritiesApi = async () => {
+  const yamlString = await fetch(AUTHORITIES_LIST_URL_MVP1).then((res) =>
+    res.text(),
+  );
+  const { authorities } = Yaml.safeLoad(yamlString);
+  if (!Array.isArray(authorities)) {
+    throw new Error('authorities yaml did not return an array of authorities');
+  }
+  return authorities;
+};
+
+const getHealthcareAuthoritiesAction = (): AppThunk<void> => async (
+  dispatch: Dispatch,
+): Promise<void> => {
+  dispatch(getHealthcareAuthorities_started());
+  try {
```

Comment:
imo its best to minimize paths occurring in reducers. Switching reducers on action types is pretty robust, when another switch is introduced (`response.kind` here), the resolution tree of the reducer becomes increasingly complex with each added condition to decipher the action.

As I understand that proposal, that would be saying 
`getHealthcareAuthoritiesResposneRecieved` with `response.kind === "success` would be equivalent to the `getHealthcareAuthorities_success` action. 

I prefer keeping actions as declarative as possible to their payloads and intent. This allows them to be agnostic to their reducers also -- If another reducer uses `getHealthcareAuthorities_success`, it can easily use that action, vs. _also_ needing to know how to infer that from a combination of action type & payload structure. 

Also, if it were to be inferred, I think that should occur in the `meta` field, so the action payloads are just the incoming data to the store state.

Reply:
you make a lot of good points! I guess i privilege having actions match as closely as possible to real world events, which in this case is that a network response was received. I see how that can be a bit more work to deal with in the dev tools though.